### ISSUE TITLE
Support for Android 13 Tiramisu (API 33)

### ISSUE THREAD

**Issue by bartekpacia** [MaikuB/flutter_local_notifications#1597]
[Android 13 introduces new runtime permission for notifications.](https://developer.android.com/about/versions/13/changes/notification-permission)

I've looked but haven't found any mention about it in the package's README on pub.dev.

So I suggest that `AndroidFlutterPluginImplementation` should get `requestPermissions` method, similar to [iOSFlutterPluginImplementation](https://pub.dev/documentation/flutter_local_notifications/latest/flutter_local_notifications/IOSFlutterLocalNotificationsPlugin/requestPermissions.html)

**Comment by MaikuB** [comment]
Thanks for raising this. Do you think could you submit a PR for this? One thing I'm also wondering is given how Android 13 appears to be in beta, I assume that making a stable release of the plugin depend on the Android 13 SDK isn't the way to go as this could cause issues for that don't have the SDK, which in turn requires a preview version of Android Studio. Is this right? If so, I suppose that support for this would require being in a pre-release so it's up to developers to opt-in with using the functionality just as they would do for using an SDK in preview?

**Comment by bartekpacia** [comment]
Hi @MaikuB 👋

Yes, I'd be very happy to submit a PR for this. Unfortunately I've encountered problems at the very beginning. Looks like Flutter (`flutter_tools`, to be precise) can't build apps that target Tiramisu.

[I've submitted an issue about this in `flutter/flutter`](https://github.com/flutter/flutter/issues/104170). I think we'll have to wait until it's resolved before we can start working on this (new notification permission) feature.

**Comment by MaikuB** [comment]
Much appreciated. Though the other thing is the SDK itself has various milestones so it may still bit too early to look at. There tends to be a beta where Google states that the APIs are final and and the SDK can be referenced by API level instead of the codename, which in itself is like a breaking change

**Comment by bartekpacia** [comment]
> There tends to be a beta where Google states that the APIs are final and and the SDK can be referenced by API level instead of the codename

Yeah, it's called _platform stability_ phase. 

![Screenshot 2022-05-19 at 1 09 01 PM](https://user-images.githubusercontent.com/40357511/169279922-b56ff137-e342-43d9-af76-cae3cc94ba7f.png)

Still, I'd like to be able to target Flutter apps on non-stable-yet platforms (i.e referenced by codenames). Let's wait until some Flutter team member answers to [that issue](https://github.com/flutter/flutter/issues/104170) and see what they say.

Summing up, I guess that the action item here is to wait until Google declares platform stability :)

**Comment by ciriousjoker** [comment]
Any updates? Platform stability has been reached.

**Comment by Cubel89** [comment]
I think API 33 is already available

**Comment by bartekpacia** [comment]
Yeah, it is.

**Comment by bartekpacia** [comment]
If no one's interested in doing this, I volunteer. I'd like to get this done over the next few days. Can you assign me @MaikuB?

**Comment by mk-dev-1** [comment]
@bartekpacia Much appreciated! Maybe this link will also be helpful here: https://blog.esper.io/android-13-exact-alarm-api-restrictions/

**Comment by stephanie-finch** [comment]
@MaikuB  Can we prioritize @bartekpacia's pull request?

Android 13 is publicly available out on Pixels now: https://www.theverge.com/2022/8/15/23306058/android-13-release-date-pixel-devices-features
